libs/api.lua

```diff
@@ -130,6 +131,44 @@ end
 
 local metaCache = {}
 
+-- Collect statistics about this server's running instance.
+-- Return these metrics in a table.  The intended consumer of this
+-- information should render it in JSON format for transmission to
+-- a time-series database and/or visualization tools.
+local function collectStats()
+  local entry, fdsUsed = 0, 0
+  local memoryUsed = 1024 * collectgarbage("count")
+
+  local function countFDs(path)
+    local entries = 0
+
+    -- You might think this leaks a filehandle, but it actually doesn't.
+    -- You can confirm this yourself by hitting /metrics several times in a row,
+    -- and observing that lua.fds.total does not increase with each GET request.
+
+    for entry in io.popen([[ls -1 "]]..path..[["]]):lines() do
+      entries = entries + 1
+    end
+    return entries
+  end
+
+  local howToDetermineFDs = {
+    Linux = function() return countFDs("/proc/self/fd") end,
+    OSX = function() return countFDs("/dev/fd") end,
+    Other = function() return -1 end
+  }
+
+  local os = ffi.os
+  if howToDetermineFDs[os] == nil then
+    os = "Other"
+  end
+
+  return {
+    ["lua.mem.used"] = memoryUsed,
```

Comment:
Could we add some more entries from libuv's functions?  The 3 memory related functions might be useful.

See this task Adam is working on for a listing of the more interesting functions. https://github.com/virgo-agent-toolkit/super-agent/issues/32